stats.go

```diff
@@ -103,6 +117,14 @@ func (g *gauge) report(name string, tags map[string]string, r StatsReporter) {
 	}
 }
 
+func (g *gauge) cachedReport() {
+	if atomic.SwapUint64(&g.updated, 0) == 1 {
+		g.cachedGauge.ReportGauge(
+			math.Float64frombits(atomic.LoadUint64(&g.curr)),
```

Comment:
Maybe same for the gauge, something that just performs the `math.Float64frombits(atomic.LoadUint64(&g.curr))`.  The compiler will likely inline it so will stay fast but is more concise to read.